### ISSUE TITLE
Custom Mean And Covariance for Variational Candidate in BNN example

### ISSUE THREAD

**Issue by BelhalK** [tensorflow/probability#985]
Hi There,

What would be the best way to access the mean and the covariance of the variational posterior distribution in the **bayesian_neural_network.py** example?

Everything is happening under the hood in `model.train_on_batch` where the `optimizer` (no matter which) computes those hyperparameters (parameters of the Gaussian variational candidate) using a descent step after computing a Monte Carlo (Flipout) approximation of the loss function (embedded in the layers of the model + cross-entropy). 

This latter MC approximation requires sampling from the variational candidate with the current values of the hyperparameters: **That's where I would like to have access to**, in order to change those hyperparameters value.

I could do it in a customer `optimizer` but was wondering if an easier way could be to do the changes right before the sampling step.

Thanks a lot.
Belhal

**Comment by BelhalK** [comment]
Actually knowing where the values of the hyperparameters are stored would fix this.
`model.get_weights()` has all the weights but what would be the syntax for the parameters of the variational candidate?

**Comment by BelhalK** [comment]
Hi there,

Say I am using SGD
```
optimizer = tf.keras.optimizers.SGD(lr=FLAGS.learning_rate)
```
Then 
```
batch_loss, batch_accuracy = model.train_on_batch(batch_x, batch_y)
```
will apply SGD step on the hyperparameters of my variational candidate.
How do I access the output mean and variance (of the SGD step). 
I feel like everything is under the hood of `train_on_batch`but I am not sure if that's the way to go.
